src/ert/gui/plottery/plot_config.py

```diff
@@ -80,12 +86,13 @@ def __init__(self, plot_settings=None, title="Unnamed", x_label=None, y_label=No
 
     def currentColor(self) -> str:
         if self._current_color is None:
-            self.nextColor()
+            return self.nextColor()
 
         return self._current_color
 
-    def nextColor(self):
-        self._current_color = next(self._line_color_cycle)
+    def nextColor(self) -> str:
+        color = next(self._line_color_cycle)
+        self._current_color = color
```

Comment:
This seems unnecessary? Why split?

Reply:
`self._current_color` has type `Optional[str]` and not `str`. However, `next(self._line_color_cycle)` will have type `str`.